relayer/pkg/chainlink/txm/txm.go

```diff
@@ -38,14 +45,14 @@ type starktxm struct {
 	queue   chan types.Transaction
 	curve   *caigo.StarkCurve
 	ks      keys.Keystore
-	cfg     starknet.Config
+	cfg     TxConfig
 
 	// TODO: use lazy loaded client
 	client    types.Provider
-	getClient func() types.Provider
+	getClient func() (types.Provider, error)
```

Comment:
Hmm, why do we again provide client like this (lazy fetch) instead of injecting it as a dependency to the txm constructor?

Reply:
so if there are multiple nodes configured, and one endpoint goes down the txm can try picking a new one